compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1712,6 +1712,20 @@ impl Context {
 
                 Ok(Self::convert_vars_to_values(vars, dfg, result_ids))
             }
+            Intrinsic::ApplyRangeConstraint => {
+                let field = self.convert_value(arguments[0], dfg).into_var()?;
+
+                let bit_size = self.convert_value(arguments[1], dfg).into_var()?;
+                let bit_size = self.acir_context.constant(bit_size).to_u128() as u32;
+
+                self.acir_context.range_constrain_var(
+                    field,
+                    &NumericType::Unsigned { bit_size },
+                    Some("call to apply_range_constraint".to_owned()),
+                )?;
+
+                Ok(Vec::new())
+            }
```

Comment:
We should be able to convert these into `Instruction::RangeCheck`s within SSA and then we can avoid these reaching ACIR gen

Reply:
```
  Intrinsic::ApplyRangeConstraint => {
            let value = arguments[0];
            let max_bit_size = dfg.get_numeric_constant(arguments[1]).unwrap().to_u128() as u32;

            SimplifyResult::SimplifiedToInstruction(Instruction::RangeCheck {
                value,
                max_bit_size,
                assert_message: Some("call to apply_range_constraint".to_owned()),
            })
        }
```

Reply:
It should be doable as we do the same for several other intrinsics.